code/modules/reagents/reagent_dispenser.dm

```diff
@@ -264,14 +266,15 @@
 	if(ZAP_OBJ_DAMAGE & zap_flags)
 		boom()
 
-/obj/structure/reagent_dispensers/fueltank/bullet_act(obj/projectile/P)
-	. = ..()
-	if(QDELETED(src)) //wasn't deleted by the projectile's effects.
-		return
```

Comment:
the comment seems to imply it wanted the inverse, are you sure this is right?

Reply:
yes because we would ALWAYS be deleted, as mentioned in the PR body. We would literally never get past line 270 EVER because of the call stack and order of operations as outlined above. the comment is six years old and is not useful, we call `qdel()` in `boom()` anyhow